src/Elliptic/BoundaryConditions/BoundaryConditionType.cpp

```diff
@@ -0,0 +1,48 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Elliptic/BoundaryConditions/BoundaryConditionType.hpp"
+
+#include <ostream>
+#include <string>
+
+#include "ErrorHandling/Error.hpp"
+#include "Options/Options.hpp"
+#include "Options/ParseOptions.hpp"
+
+/// \cond
+namespace elliptic {
+std::ostream& operator<<(
+    std::ostream& os,
+    const elliptic::BoundaryConditionType boundary_condition_type) noexcept {
+  switch (boundary_condition_type) {
+    case elliptic::BoundaryConditionType::Dirichlet:
+      os << "Dirichlet";
+      break;
+    case elliptic::BoundaryConditionType::Neumann:
+      os << "Neumann";
+      break;
+    default:
+      ERROR("Invalid case");
```

Comment:
say what is invalid (the type)

Reply:
[nitpicking on this, not a change request] if this is error is encountered, isn't it most likely because someone added a new enum value and didn't correspondingly adjust the op<<? In other words, the type isn't invalid, but the op<< is incomplete. I feel the most useful message, given this, would be something like "Missing a case for operator<<(the type)"